### ISSUE TITLE
API New: User-created images

### ISSUE THREAD

**Issue by achimnol** [lablup/backend.ai-manager#119]
Let's users be able to stop & commit their sessions as a new Docker image, and reuse them afterwards.

At first, we will support this only in single-container sessions.

**Comment by achimnol** [comment]
In 19.03.0b series, now we have automatic digest-based image pulling in the agent side with a full integration with public/private Docker registries.

The next steps:
* [ ] Ability to snapshot a live container
* [ ] Ability to push the snapshotted image to a local registry

**Comment by achimnol** [comment]
* If we allow dot-prefixed vfolders (e.g., `.local`), users will be able to keep their user-specific installation of Python packages across different compute sessions. #144 enables this.
 * For security, we are now moving forward to disallow root inside containers in all setups. In this sense, we need a new way to customize the docker images.
   - #171 allows importing an external Docker image with some constraints.
   - We could use [userspace image build tools](https://github.com/lablup/backend.ai-manager/pull/171#issuecomment-517541125) to allow rootless custom image build operation.

**Comment by achimnol** [comment]
We have deployed a partially generalized solution as #171.
Currently our following approaches render the necessity of having container snapshots:
 * We don't allow root inside containers.
 * We let users install their own user-packages in the user directory, which is mounted from the host-side scratch directory.
 * We provide several ways to keep the user-specific content across different compute sessions by lablup/backend.ai#98 and lablup/backend.ai#99.

Closing.
